lib/_parse-drill.js

```diff
@@ -208,4 +193,59 @@ var parse = function(parser, block) {
   return
 }
 
-module.exports = parse
+var flush = function(parser) {
+  if (parser._drillStash.length) {
+    parser._drillStash.forEach(function(data) {
+      if (!parser.format.zero && reCOORD.test(data.block)) {
+        parser.format.zero = 'T'
+        parser._warn('zero suppression missing and not detectable;'
+          + ' assuming trailing suppression')
+      }
+      parseBlock(parser, data.block, data.line)
+    })
+    parser._drillStash = []
+  }
+}
+
+var parse = function(parser, block) {
+  parser._drillStash = parser._drillStash || []
+
+  // parse comments for formatting hints and ignore the rest
+  if (block[0] === ';') {
+    // check for kicad format hints
+    var formatHints = parseCommentForFormatHints(parser, block, parser.line)
+
+    Object.keys(formatHints).forEach(function(key) {
+      if (!parser.format[key]) {
+        parser.format[key] = formatHints[key]
+      }
+    })
+
+    return
+  }
+
+
+  // detect or assume zero suppression
+  if (!parser.format.zero) {
+    if (parser._drillStash.length >= 1000) {
+      flush(parser)
+    }
+    else {
+      if (reCOORD.test(block)) {
```

Comment:
I'm pretty sure this stash logic is too greedy. We'll completely miss stuff `INCH,TZ` where we don't need to do detection anymore. Also, until we see something with a coordinate that we can't parse because of missing format, there's no need to stash anything at all. This explains why the tests need so many `p.end()`s.